include/api/Metrics.h

```diff
@@ -289,10 +280,126 @@ class Metrics
 
   mutable std::mutex _mutex;
   LookupTable _lookups;
-  MetricBlobs _blobs;
+  BlobStorage _blobs;
   uint16_t _cur_blob = 0;
   uint16_t _cur_off  = 0;
 
+public:
+  // These are sort of factory classes, using the Metrics singleton for all storage etc.
+  class Gauge
+  {
+  public:
+    using self_type = Gauge;
+    using SpanType  = Metrics::SpanType;
+
+    class AtomicType : public Metrics::AtomicType
+    {
+    };
```

Comment:
Why not just `using AtomicType = Metrics::AtomicType;` ?

Reply:
Because then the compiler won't give errors or warnings about wrong types.

Reply:
Why don't you give it the constructor `explicit AtomicType(Metrics::AtomicType a) : Metrics::AtomicType(a) {}` so you won't need the reinterpret_cast.

Reply:
Ok, let me look tomorrow morning.